In [3]:
import requests
brackets = requests.get("http://adventofcode.com/day/1/input").text
print (brackets)

Puzzle inputs differ by user.  Please log in to get your puzzle input.



Waa. That's disappointing. Let's try snaffling the cookie from the browser window: (in Chrome) right click, <kbd>Inspect Element</kbd>, <kbd>Network</kbd>, <kbd>F5</kbd> to refresh, click the file, <kbd>Headers</kbd>. I've copy-pasted it into a cookie.py file that's something like `cookie = "session=..."`

In [6]:
import cookie
print (cookie.cookie[:8] + '...')

session=...


In [16]:
import requests
cookie_crumbs = cookie.cookie.partition('=') # ("session", "=", "...")
cookie_dict = {cookie_crumbs[0]: cookie_crumbs[2]}
brackets=requests.get("http://adventofcode.com/day/1/input", cookies=cookie_dict).text
print (brackets[:10], len(brackets))

()()(()()( 7000


That's a lot of brackets. Time to count them!

In [22]:
left = len(['(' for x in brackets if x == '('])
right = len([')' for x in brackets if x == ')'])
print (left, right, left-right)

3640 3360 280


Yay; 280 is the right answer for me. Now I need a solution that finds the first negative basement; think I actually need a proper loop now.

In [26]:
floor = 0
for i, bracket in enumerate(brackets):
    if bracket == "(":
        floor = floor + 1
    if bracket == ")":
        floor = floor - 1
    if floor < 0:
        print ("Entering basement at bracket {}".format(i))
        break
        
    

Entering basement at bracket 1796


Gah. That's off by one because Python starts at 0 and the question starts at 1. :(

But yay, 1797 is my right answer.